File: VB/WinSolution3.Module/DomainObject1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.ExpressApp.Editors

Namespace WinSolution3.Module
	<DefaultClassOptions> _
	Public Class DomainObject1
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _Name, value)
			End Set
		End Property
		Private _Description As String
		Public Property Description() As String
			Get
				Return _Description
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Description", _Description, value)
			End Set
		End Property
		Private _Value As Integer
		Public Property Value() As Integer
			Get
				Return _Value
			End Get
			Set(ByVal value As Integer)
				SetPropertyValue("Value", _Value, value)
			End Set
		End Property
		<Association("DomainObject1-DomainObject2s")> _
		Public ReadOnly Property DomainObject2s() As XPCollection(Of DomainObject2)
			Get
				Return GetCollection(Of DomainObject2)("DomainObject2s")
			End Get
		End Property
	End Class
End Namespace

File: CS/WinSolution3.Module/DomainObject1.cs

using DevExpress.Xpo;

using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.ExpressApp.Editors;

namespace WinSolution3.Module {
    [DefaultClassOptions]
    public class DomainObject1 : BaseObject {
        public DomainObject1(Session session)
            : base(session) { }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
        private string _Description;
        public string Description {
            get { return _Description; }
            set { SetPropertyValue("Description", ref _Description, value); }
        }
        private int _Value;
        public int Value {
            get { return _Value; }
            set { SetPropertyValue("Value", ref _Value, value); }
        }
        [Association("DomainObject1-DomainObject2s")]
        public XPCollection<DomainObject2> DomainObject2s {
            get {
                return GetCollection<DomainObject2>("DomainObject2s");
            }
        }
    }
}
